crates/db_schema/src/source/person.rs

```diff
@@ -135,36 +133,49 @@ pub struct PersonUpdateForm {
   pub ban_expires: Option<Option<DateTime<Utc>>>,
 }
 
-#[derive(PartialEq, Eq, Debug)]
+#[skip_serializing_none]
+#[derive(Clone, PartialEq, Eq, Debug, Serialize, Deserialize)]
 #[cfg_attr(
   feature = "full",
-  derive(Identifiable, Queryable, Selectable, Associations)
+  derive(Identifiable, Queryable, Selectable, Associations, TS)
 )]
 #[cfg_attr(feature = "full", diesel(belongs_to(crate::source::person::Person)))]
 #[cfg_attr(feature = "full", diesel(table_name = person_actions))]
 #[cfg_attr(feature = "full", diesel(primary_key(person_id, target_id)))]
 #[cfg_attr(feature = "full", diesel(check_for_backend(diesel::pg::Pg)))]
-pub struct PersonFollower {
-  #[cfg_attr(feature = "full", diesel(column_name = target_id))]
+#[cfg_attr(feature = "full", ts(export))]
+pub struct PersonActions {
+  pub target_id: PersonId,
   pub person_id: PersonId,
-  #[cfg_attr(feature = "full", diesel(column_name = person_id))]
-  pub follower_id: PersonId,
-  #[cfg_attr(feature = "full", diesel(select_expression = person_actions::followed.assume_not_null()))]
-  #[cfg_attr(feature = "full", diesel(select_expression_type = dsl::AssumeNotNull<person_actions::followed>))]
-  pub published: DateTime<Utc>,
-  #[cfg_attr(feature = "full", diesel(select_expression = person_actions::follow_pending.assume_not_null()))]
-  #[cfg_attr(feature = "full", diesel(select_expression_type = dsl::AssumeNotNull<person_actions::follow_pending>))]
-  pub pending: bool,
+  #[cfg_attr(feature = "full", ts(optional))]
+  /// When the person was followed.
+  pub followed: Option<DateTime<Utc>>,
+  #[cfg_attr(feature = "full", ts(optional))]
+  /// Whether the follow is pending.
+  pub follow_pending: Option<bool>,
```

Comment:
Lemmy doesnt allow following persons so this is only relevant for federation and can use `serde(skip)` (or not even select it from the db).

Reply:
K, I'll add that for both these columns.